File: LeetCodeSolutions/LeetCode_0125.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.upper()
        tmp = ''
        for _ in s:
            if '0' <= _ <= '9' or 'A' <= _ <= 'Z':
                tmp += _
        # n = len(tmp)
        i, j = 0, len(tmp) - 1
        while i < j:
            if tmp[i] != tmp[j]:
                return False
            i += 1
            j -= 1
        return True

File: LeetCodeSolutions/LeetCode_0125.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string ss;
        for(auto c: s){
            if ((c >= 'a' && c <= 'z')||(c >= 'A' && c <= 'Z')||(c >='0' && c<='9')) ss += c;
        }
        int i = 0, j = ss.size() - 1;
        while (i < j){
            auto a = ss[i], b = ss[j];
            if (a >= 'A' && a <= 'Z') a = 'a' + a - 'A';
            if (b >= 'A' && b <= 'Z') b = 'a' + b - 'A';
            if (a != b) return false;
            ++i;
            --j;
        }
        return true;
    }
};